modular/blocksyncer/blocksyncer_indexer.go

```diff
@@ -1,551 +0,0 @@
-		} else if storageprovider.StorageProviderEvents[event.Type] {
-			spEvent = append(spEvent, e)
-		} else if virtualgroup.VirtualGroupEvents[event.Type] {
-			VirtualGroupEvent = append(VirtualGroupEvent, e)
-		} else if spExit.SpExitEvents[event.Type] {
-			exitEvent = append(exitEvent, e)
-		} else if event.Type == object.EventDeleteObject || event.Type == object.EventCreateObject || payment.PaymentEvents[event.Type] || virtualgroup.VirtualGroupEvents[event.Type] {
-			data, err := i.ExtractEvent(ctx, block, common.Hash{}, e.Event)
-			if err != nil {
-				return err
-			}
-			dataList = append(dataList, data)
-			objectIDEvent = append(objectIDEvent, e)
-			prefixEvent = append(prefixEvent, e)
-		} else if object.ObjectEvents[event.Type] {
-			objectEvent = append(objectEvent, e)
-		}
-		if prefixtree.BuildPrefixTreeEvents[event.Type] {
-			prefixEvent = append(prefixEvent, e)
-		}
-	}
-
-	if len(dataList) > 0 {
-		err := i.BatchHandle(ctx, dataList, block, common.Hash{})
-		if err != nil {
-			return err
-		}
-	}
-
-	allEvents := make([][]TxHashEvent, 0)
-	allEvents = append(allEvents, bucketEvent)
-	allEvents = append(allEvents, groupEvent)
-	allEvents = append(allEvents, objectEvent)
-	allEvents = append(allEvents, permissionEvent)
-	allEvents = append(allEvents, spEvent)
-	allEvents = append(allEvents, prefixEvent)
-	allEvents = append(allEvents, VirtualGroupEvent)
-	allEvents = append(allEvents, exitEvent)
-	allEvents = append(allEvents, objectIDEvent)
-
-	return i.concurrenceHandleEvent(ctx, block, allEvents)
-}
-
-// HandleGenesis accepts a GenesisDoc and calls all the registered genesis handlers in the order in which they have been registered.
-func (i *Impl) HandleGenesis(genesisDoc *tmtypes.GenesisDoc, appState map[string]json.RawMessage) error {
-	return nil
-}
-
-// HandleBlock accepts block and calls the block handlers.
-func (i *Impl) HandleBlock(block *coretypes.ResultBlock, events *coretypes.ResultBlockResults, txs []*types.Tx, getTmcValidators modules.GetTmcValidators) {
-	for _, module := range i.Modules {
-		if blockModule, ok := module.(modules.BlockModule); ok {
-			err := blockModule.HandleBlock(block, events, txs, getTmcValidators)
-			if err != nil {
-				log.Errorw("error while handling block", "module", module.Name(), "height", block.Block.Height, "err", err)
-			}
-		}
-	}
-}
-
-// HandleTx accepts the transaction and calls the tx handlers.
-func (i *Impl) HandleTx(tx *types.Tx) {
-	log.Info("HandleTx")
-}
-
-// HandleMessage accepts the transaction and handles messages contained inside the transaction.
-func (i *Impl) HandleMessage(block *coretypes.ResultBlock, index int, msg sdk.Msg, tx *types.Tx) {
-	log.Info("HandleMessage")
-}
-
-// Processed tells whether the current Indexer has already processed the given height of Block
-// An error is returned if the operation fails.
-func (i *Impl) Processed(ctx context.Context, height uint64) (bool, error) {
-	ep, err := i.DB.GetEpoch(context.Background())
```

Comment:
Why not use the field  `ProcessedHeight` of `Impl`, get from db seems unnecessary.

Reply:
We need to discuss that